### ISSUE TITLE
Unexpected requests via IPv6

### ISSUE THREAD

**Issue by MaxKh** [rclone/rclone#5436]
#### What is the problem you are having with rclone?
Sometimes rclone tries to make requests via IPv6 although IPv6 is disabled globally with ipv6.disable=1 kernel option

#### What is your rclone version (output from `rclone version`)
rclone v1.53.3-DEV
- os/arch: linux/arm
- go version: go1.16.5

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
rclone in docker container based on alpinelinux 3.14 armhf

HOST: ODRIOD-XU4 - Official image with Ubuntu 18.04.5 LTS
Kernel: 4.14.180-178
Docker: 20.10.2, build 20.10.2-0ubuntu1~18.04.2

#### Which cloud storage system are you using? (e.g. Google Drive)
Yandex

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
copy or sync

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
` Failed to copy: Put "https://uploader14g.disk.yandex.net:443/upload-target/20210704T202011.388.utd.cofbc2zzvmx0l3xk3iwx3zp5i-k14g.2442071": dial tcp [2a02:6b8:c03:762:0:564:ddd0:71e3]:443: socket: address family not supported by protocol`

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by MaxKh** [comment]
May be related to
https://github.com/golang/go/issues/25321
https://github.com/golang/go/issues/45024

**Comment by ncw** [comment]
This is unlikely to be an rclone bug more a go standard library bug.

> May be related to
> [golang/go#25321](https://github.com/golang/go/issues/25321)
> [golang/go#45024](https://github.com/golang/go/issues/45024)

Those seem to be about the DNS resolver connections

You should be able to work-around with `--bind 0.0.0.0` though

Also compiling your own rclone from scratch will probably fix the problem too.

**Comment by MaxKh** [comment]
> You should be able to work-around with --bind 0.0.0.0 though

Got the same error
`Failed to copy: Put "https://uploader7o.disk.yandex.net:443/upload-target/20210706T083053.929.utd.7rjb2d56tbpz7685lr7kff4vl-k7o.9966166": dial tcp 0.0.0.0:0->[2a02:6b8:c02:c19:0:564:f7d6:a910]:443: socket: address family not supported by protocol`

**Comment by ncw** [comment]
> > You should be able to work-around with --bind 0.0.0.0 though
> 
> Got the same error
> `Failed to copy: Put "https://uploader7o.disk.yandex.net:443/upload-target/20210706T083053.929.utd.7rjb2d56tbpz7685lr7kff4vl-k7o.9966166": dial tcp 0.0.0.0:0->[2a02:6b8:c02:c19:0:564:f7d6:a910]:443: socket: address family not supported by protocol`

Hmm...

Did you try the compile your rclone from scratch? This should make rclone use the system resolver.

**Comment by MaxKh** [comment]
> This should make rclone use the system resolve

Should I set CGO_ENABLED=1 here?
https://git.alpinelinux.org/aports/tree/community/rclone/APKBUILD

**Comment by ncw** [comment]
> > This should make rclone use the system resolve
> 
> Should I set CGO_ENABLED=1 here?
> https://git.alpinelinux.org/aports/tree/community/rclone/APKBUILD

Yes that would be worth a try.

**Comment by MaxKh** [comment]
I've been testing my build with CGO_ENABLED=1 for several days - no problems so far

**Comment by ncw** [comment]
OK that is good to know. It seems that the Go resolver doesn't understand your system properly for some reason...

There is a plan in https://github.com/rclone/rclone/issues/5090 to provide an official CGO build for rclone which would fix this problem.

**Comment by ivandeex** [comment]
OS resolver support will be covered by the task developing cgo-based rclone package.

**Comment by animosity22** [comment]
This one is pretty strange. I have a Linux box with IPV6 off (there is no IPV6 on Verizon FIOS) and with the --bind command and I got a few of these here/there

```
Post "https://content.dropboxapi.com/2/files/download": dial tcp 0.0.0.0:0->[2620:100:6019:14::a27d:40e]:443: socket: address family not supported by protocol

```

My mount has
```
--bind 0.0.0.0
```
tagged on it so wonder why it's IPV4 is trying to get hit the IPV6.

**Comment by ncw** [comment]
I've merged a workaround for this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.64 

You can use `--bind 0.0.0.0` to force rclone to use IPv4 addresses and `--bind ::0` to force rclone to use IPv6 addresses.
